extensions/Worker.Extensions.Http.AspNetCore/src/FunctionsMiddleware/FunctionsHttpProxyingMiddleware.cs

```diff
@@ -85,6 +85,9 @@ private static async Task<bool> TryHandleHttpResult(object? result, FunctionCont
                     // processing is required.
                     context.GetInvocationResult().Value = null;
                     break;
+                case AspNetCoreHttpResponseData when !isInvocationResult:
```

Comment:
I wonder if we should test for the base class - `HttpResponseData` instead? In theory, I could see a customer adding a middleware which will replace/wrap that input binding with their own. We would still want this to work in that case.

Now I am unsure if that scenario is possible with todays extension points. But unless we absolutely need the type to be exactly `AspNetCoreHttpResponseData`, it is typically better to work off the base class.

Reply:
The above case for the invocation result also uses the more specific type (`AspNetCoreHttpResponseData`). The reasoning is captured [here](https://github.com/Azure/azure-functions-dotnet-worker/pull/2322#discussion_r1545489527) in another code review. I also had the same thought originally.